src/pages/index.js

```diff
@@ -140,39 +189,102 @@ const addCardPopup = new PopupWithForm(
 addCardPopup.setEventListeners();
 
 // Add Card Button
+const addCardButton = document.querySelector("#add-card-button");
+
 addCardButton.addEventListener("click", () => {
-  // Reset the Input fields on the add card popup
   addCardFormValidator.resetValidation();
 
   addCardPopup.open();
 });
 
+// Create a new card
 function handleCardSaveSubmit(inputValues) {
-  // Create a new card instance
-  const newCardInstance = renderCard({
+  const saveButton = document.getElementById("add-card-save-button");
+
+  saveButton.textContent = "Saving...";
+
+  const cardInfo = {
     name: inputValues.title,
     link: inputValues.url,
-  });
+  };
+
+  api
+    .addNewCard(cardInfo)
+    .then((res) => {
+      console.log("New Card:", cardInfo);
+      renderCard(res);
+      addCardPopup.close();
+    })
+    .catch((error) => {
+      console.error("API Error:", error);
+    })
+    .finally(() => {
+      saveButton.textContent = "Save";
+    });
 }
 
-// ************************ Inital Card Setup ************************ //
+// *************************** Delete Card  ************************** //
+const confirmDeletePopup = new PopupWithSubmit({
+  popupSelector: "#delete-card-modal",
+});
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-    handFormSubmit: (item) => {
-      renderCard(item);
-    },
-  },
-  "#cards-list" // container selector
+const confirmDeleteButton = document.querySelector(
+  "#confirm-delete-modal-button"
 );
 
-cardSection.renderItems();
+confirmDeletePopup.setEventListeners();
+
+function handleDeleteIconClick(card) {
+  confirmDeletePopup.open(card);
+  confirmDeleteButton.focus();
+  confirmDeletePopup.setSubmit(() => {
+    card.deleteCard();
+    confirmDeletePopup.close();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.